app/controllers/reviews_controller.rb

```diff
@@ -20,10 +20,15 @@ def edit
   end
 
   def update
+    @shelter = Shelter.find(params[:shelter_id])
     review = Review.find(params[:id])
     review.update(review_params)
-    review.save
-    redirect_to "/shelters/#{params[:shelter_id]}"
+    if review.save
+      redirect_to "/shelters/#{params[:shelter_id]}"
+    else
+      flash[:notice] = "Review not updated: Reviews must have a title, rating, and content"
+      redirect_to "/shelters/#{@shelter.id}/reviews/#{review.id}/edit"
```

Comment:
I hard coded (think thats the way to refer to this) the redirect rather than using `render :new` here because it was redirecting to the create review page for some reason, this seems to work both in tests and on local server.

Reply:
I think you might be able to use render :edit but I'm just guessing. I will try this on my computer and push up a commit to this branch if I can.